File: ABC/abc326/src/c.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        m: usize,
        mut a: [usize; n],
    }
    a.sort();

    let mut res = 0;
    let mut r = 0;
    for l in 0..n {
        while r < n && a[r as usize] < a[l] + m {
            r += 1;
        }
        res = res.max(r - l);
    }

    println!("{}", res);
}

File: abc/abc326/c.py
from bisect import bisect_left

N, M = map(int, input().split())
A = list(map(int, input().split()))
A.sort()

ans = 0
for i, a in enumerate(A):
    ans = max(ans, bisect_left(A, a + M) - i)

print(ans)
